includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -629,6 +629,12 @@ public function register_scripts() {
 		);
 
 		wp_set_script_translations( 'WCPAY_CHECKOUT', 'woocommerce-payments' );
+
+		if ( ! WC()->cart->needs_payment() && is_checkout() && ! has_block( 'woocommerce/checkout' ) ) {
```

Comment:
Nitpick: moving `is_checkout()` as the first check as it can help terminate this code for most cases, which results in a little bit faster processing in general.